cdap-ui/server/express.js

```diff
@@ -161,7 +164,17 @@ function makeApp (authAddress, cdapConfig) {
   // any other path, serve index.html
   app.all('*', [
     function (req, res) {
-      res.sendFile(DIST_PATH + '/index.html');
+      // BCookie is the browser cookie, that is generated and will live for a year.
+      // This cookie is always generated to provide unique id for the browser that 
+      // is being used to interact with the CDAP backend.
+      if(! req.cookies.bcookie) {
+        var date = new Date();
+        date.setDate(date.getDate() + 365); // Expires after a year.
+        res.cookie('bcookie', uuid.v4(), { expires: date});
+      } else {
+        res.cookie('bcookie', req.cookies.bcookie, { expires: date});
```

Comment:
``date`` seems undefined here, no?

bcookie intrigued me, so checked this PR :)